Firestore/core/src/local/leveldb_persistence.cc

```diff
@@ -178,20 +178,30 @@ util::Status LevelDbPersistence::ClearPersistence(
   return fs->RecursivelyRemove(leveldb_dir);
 }
 
-int64_t LevelDbPersistence::CalculateByteSize() {
+util::StatusOr<int64_t> LevelDbPersistence::CalculateByteSize() {
   auto* fs = Filesystem::Default();
 
   int64_t count = 0;
   auto iter = util::DirectoryIterator::Create(directory_);
   for (; iter->Valid(); iter->Next()) {
-    int64_t file_size = fs->FileSize(iter->file()).ValueOrDie();
+    StatusOr<int64_t> maybe_size = fs->FileSize(iter->file());
+    if (!maybe_size.ok()) {
+      return Status::FromCause("Failed to size LevelDB directory",
+                               maybe_size.status());
+    }
+
+    int64_t file_size = maybe_size.ValueOrDie();
     count += file_size;
   }
 
-  HARD_ASSERT(iter->status().ok(), "Failed to iterate LevelDB directory: %s",
-              iter->status().error_message().c_str());
-  HARD_ASSERT(count >= 0 && count <= std::numeric_limits<int64_t>::max(),
-              "Overflowed counting bytes cached");
+  if (!iter->status().ok()) {
+    return Status::FromCause("Failed to iterate over LevelDB files",
+                             iter->status());
+  }
+  if (count < 0) {
```

Comment:
I'm not sure about this check. `int` overflow is undefined behavior, so it's not guaranteed to work. Also, technically it should be an assertion (the behavior of the whole program is undefined from this point). OTOH, I guess this is likely to work in practice, so feel free to keep as is (it's not like we have a great alternative).

Reply:
PTAL I've adjusted this to use an unsigned accumulator and then check for overflow on each iteration.

Reply:
LGTM.